src/Core/Managed/Shared/OperationTelemetryExtensions.cs

```diff
@@ -16,9 +16,12 @@ public static class OperationTelemetryExtensions
         /// </summary>
         /// <param name="telemetry">Telemetry item object that calls this extension method.</param>
         public static void Start(this OperationTelemetry telemetry)
-        {
-            var startTime = DateTimeOffset.UtcNow;
-            telemetry.Timestamp = startTime;
+        {            
+            telemetry.Timestamp = DateTimeOffset.UtcNow;
+
+            // Begin time is used internally for calculating duration of operation at the end call,
+            // and hence is stored using higher precision Clock.
+            telemetry.BeginTime = Clock.Instance.Time;
```

Comment:
If this is the only place that Clock is used do we really need a class for it or should the singleton Stopwatch instance just go someplace more direct?